compiler/rustc_mir_build/src/thir/pattern/usefulness.rs

```diff
@@ -884,6 +884,9 @@ fn collect_nonexhaustive_missing_variants<'p, 'tcx>(
     cx: &MatchCheckCtxt<'p, 'tcx>,
     column: &[&DeconstructedPat<'p, 'tcx>],
 ) -> Vec<WitnessPat<'tcx>> {
+    if column.is_empty() {
+        return Vec::new();
+    }
     let ty = column[0].ty();
```

Comment:
this could have been `get` + let-else
```rust
    let Some(ty) = column.get(0) else { return Vec::new(); };
    let ty = ty.ty();
```
not that it matters though.